### ISSUE TITLE
Is there anyway to set the starting address of global/static data area?

### ISSUE THREAD

**Issue by xbcnn** [emscripten-core/emscripten#17936]
Does anyone know if we can do this in Emscripten?  I mean to keep a no man's zone before it, such as 64KB(a page).

**Comment by sbc100** [comment]
If you mean the start of the heap, then you can use address of the linker-generated symbol `__heap_base`... or you can use  `sbrk(0)` to find out the current end of the heap.

If you mean the static data area then you can use the the address of the `__global_base`.

**Comment by xbcnn** [comment]
> If you mean the static data area then you can use the the address of the `__global_base`.

Yeah, I mean this.  Is it possible to change it, like by passing some linker options?

**Comment by sbc100** [comment]
> > If you mean the static data area then you can use the the address of the `__global_base`.
> 
> Yeah, I mean this. Is it possible to change it, like by passing some linker options?

Yes, you can build with `-sGLOBAL_BASE=66560`

**Comment by xbcnn** [comment]
Thank you @sbc100. It's nice.

**Comment by xbcnn** [comment]
@sbc100  One more question.  Is there any way to detect load/storing to this no-man zone before the static data area?

**Comment by sbc100** [comment]
We do support `-fsanitize=address` which will detect any out-of-bounds access.

Also, if you build with `-sASSERTIONS` that will enabled `-sSTACK_OVERFLOW_CHECK` which will write magic values that we write to the very beginning of memory and these get checked at various points.

What exactly are you trying to achieve here?

Also, just FYI, I just made a change that, by default puts the stack first in debug builds, which then means that static data will come after the stack region in memory. See #18154
